src/Symfony/Component/Debug/ExceptionHandler.php

```diff
@@ -412,19 +412,12 @@ private function formatArgs(array $args)
 
     /**
      * Returns an UTF-8 and HTML encoded string
+     *
+     * @deprecated since version 2.7, to be removed in 3.0.
      */
     protected static function utf8Htmlize($str)
     {
-        if (!preg_match('//u', $str) && function_exists('iconv')) {
-            set_error_handler('var_dump', 0);
-            $charset = ini_get('default_charset');
-            if ('UTF-8' === $charset || $str !== @iconv($charset, $charset, $str)) {
-                $charset = 'CP1252';
-            }
-            restore_error_handler();
-
-            $str = iconv($charset, 'UTF-8', $str);
-        }
```

Comment:
Two reasons: because of the `function_exists` check, the behavior is not deterministic, and not all strings are convertible from CP1252 to UTF-8, (`"\x9D"`). The code block is broken. Because of the function_exists check, this can be removed without BC pb